taskograph.Web/Controllers/TaskController.cs

```diff
@@ -71,12 +81,141 @@ public IActionResult AddEntry(int taskId, long minutes)
 
         public IActionResult ConfigTasks()
         {
-            string _userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            string _userId = GetIdentityUserId();
             ConfigTasksViewModel configTasksVM = new ConfigTasksViewModel();
             configTasksVM.Tasks = _taskRepository.GetAllTaskDTOs(_userId).ToList();
             configTasksVM.Groups = _groupRepository.GetAll(_userId).ToList();
 
             return View("ConfigTasks", configTasksVM);
         }
+
+        public IActionResult AddTask()
+        {
+            string _userId = GetIdentityUserId();
+            TaskViewModel taskVM = new TaskViewModel();
+            ReadGroupsSelectedItems(taskVM);
+            ReadColorsSelectedItems(taskVM);
+            taskVM.IsFormForTask = true;
+            return View("AddTask", taskVM);
+        }
+
+       
+
+        [HttpPost]
+        public IActionResult AddTask(TaskViewModel taskVM)
+        {
+            string _userId = GetIdentityUserId();
+            Task task = new Task()
+            {
+                Name = taskVM.Name,
+                GroupId = (taskVM.GroupId == UNASSIGNED_INT ? null : taskVM.GroupId),
+                ColorId = taskVM.ColorId,
+                AppUserId = _appUserRepository.GetId(_userId)
+            };
+            _taskRepository.Add(task);
+            return ConfigTasks();
+        }
+
+        public IActionResult AddGroup()
+        {
+            string _userId = GetIdentityUserId();
+            TaskViewModel taskVM = new TaskViewModel();
+            ReadColorsSelectedItems(taskVM);
+            ReadTasksSelectedItems(taskVM);
+            taskVM.IsFormForTask = false;
+            return View("AddTask", taskVM);
```

Comment:
Jeśli widok służy do dodawania zadania ORAZ grupy, to może bardziej adekwatnie będzie nazwać go "Add"?